### ISSUE TITLE
send amount should be 0 by default

### ISSUE THREAD

**Issue by wayfu** [vsariola/sointu#178]
way too many parameters can create very undesirable effects if 64 are suddenly added to them :)

**Comment by vsariola** [comment]
64 = 0.0, 0 = -1.0 and 128 = 1.0. So, the default *is* 0.0f :)

**Comment by wayfu** [comment]
default should be 0 effect, 128 is +64

**Comment by vsariola** [comment]
I think it's a bit gotcha to not get any effect whatsoever, would 96 (0.5) be a good compromise? So you can here that something is happening, but it's not as over the top as 128? I'm a bit worried that users first time trying out send will try targeting something and nothing happens if the amount is set to 64 (0.0f) by default.

**Comment by wayfu** [comment]
is that your target group?

**Comment by vsariola** [comment]
I went through Ronny's song and Ronny used send amount values of 65, 112, 75, 72, 75, 128, 23, 122, 65, 128, 99, 101, 81, 91. Meanwhile, Virgill's latest song used send amount values of 80, 128, 58, 128, 128, 74, 72, 80, 128, 128.

The median of all those comes down as 86. But 80 (16+64) or 96 (32+64) should compress better. So defaulting to either 80 or 96 makes far more sense to me than 64 which is a value that is never used.

**Comment by wayfu** [comment]
i'm thinking about this from the musician's perspective. no send will end up at 64 because that does nothing, but it's a good starting point because it allows us to dial in the effect of it from zero. this is how it works in the big vst synths like massive, serum, etc as well.

**Comment by vsariola** [comment]
I'm thinking this from a project maintainers perspective. If I make send default to 64, there will be people opening issues why the send does nothing, even though they have defined a target for the modulation. I could a) make it snaller, but not 0.0 (e.g. 80); and b) make the unit defaults user-configurable so everyone can adjust them how they please

**Comment by vsariola** [comment]
The key here is that many don't realize 64 means 0; they think it's 0.5. I've already had to explain this to new users. So they might think with 64 they have 0.5 as the send gain and then just wonder why nothing happens

**Comment by qm210** [comment]
I was thinking, a button for "reset parameter to default value" would be a nice thing in general. could that help solve the confusion here, too?

as in, if I press the "default" button on "amount", as a musician I also would expect that whatever value it now has, it means "no send"

**Comment by vsariola** [comment]
There was a way to "reset to default value" but I broke it when refactoring (rewriting the GUI). But I am still not convinced that user expects the default of send to have no effect (amount = 64); I think that sounds like what you would expect to happen when you delete the send unit altogether.

**Comment by wayfu** [comment]
i find myself setting the send value to 64 (0) before i assign it *every time* i use a send. first of all, when selecting a target, the send inevitably, because of the way the assignment works, sends to a couple of places it isn't meant to go (the first op of the instrument you chose, and then the ports in the selected op that you pass before getting to the one you want) - where it may actually wreak havoc, especially with extreme settings like 128. even on the correct target, it may do that, or create noise bursts (filter frequency > 150 or so, etc). after a few very annoying instances of this i now turn the send amount down before i assign the send.

**Comment by wayfu** [comment]
> The key here is that many don't realize 64 means 0; they think it's 0.5. I've already had to explain this to new users. So they might think with 64 they have 0.5 as the send gain and then just wonder why nothing happens

maybe then change the GUI to say 0 (raw value: 64/ 0x40)

**Comment by vsariola** [comment]
I agree that the way the port is selected could be improved, as suggested by distance in #51. If the GUI was the way distance suggested, you would least you would not be sending to random ports on the way finding the correct port.

**Comment by wayfu** [comment]
yeah that would be a good change, i think. in addition to default to 0 effect (raw value 64) on the send!

**Comment by vsariola** [comment]
Both changes now made their way to the master branch: the Send amount is 0 by default and they way you select the port is by clicking the parameter (port) that you want to modulate. The send knob still shows 64 as the value, but the knob showing negative and positive side and the dial being in the middle gives the user a hint that the value is 0. Also, the tooltip shows the floating point value once you hover on it.
